File: grises.py
import cv2

fuente = 	cv2.imread('puppy.jpg')
copia = cv2.cvtColor(fuente, cv2.COLOR_BGR2GRAY)

#cv2.imwrite('./pics/gris-cs50.jpg', img_grises)

cv2.imshow('original', fuente)
cv2.imshow('grises', copia)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: cpp/grises.cpp
#include <opencv2/opencv.hpp>
using namespace cv;

int main( int argc, char** argv )
{
	Mat origen;

	// Leer imagen
 	origen = imread("puppy.jpg", IMREAD_COLOR );

	if(!origen.data )
	{
   	printf( "No existe la imagen\n " );
   	return -1;
	}

	Mat destino;
	cvtColor(origen, destino, COLOR_BGR2GRAY); // convertido a grises

	//namedWindow( origen, WINDOW_AUTOSIZE );
	imshow( "Original", origen );
	imshow( "Grises", destino );

	waitKey(0);

	return 0;
}
